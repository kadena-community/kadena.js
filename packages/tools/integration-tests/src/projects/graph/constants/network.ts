

export const baseUrl = 'http://localhost:4000/graphql'
