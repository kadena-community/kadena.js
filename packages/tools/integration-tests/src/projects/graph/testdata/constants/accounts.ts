import type { ChainId } from "@kadena/types";

export interface IAccount {
  account: string,
  publicKey: string,
  chainId: ChainId,
  guard: string,
  secretKey: string,
}

// the pre-funded account that will be used to fund other accounts
export const sender00Account: IAccount = {
  account: 'sender00',
  publicKey: '368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca',
  chainId: '0',
  guard: '368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca',
  // this is here only for testing purposes. in a real world scenario, the secret key should never be exposed
  secretKey: '251a920c403ae8c8f65f59142316af3c82b631fba46ddea92ee8c95035bd2898',
};

export const accountOne: IAccount = {
  account: `k:bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8`,
  publicKey: "bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8",
  chainId: '0',
  guard: "bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8",
  // this is here only for testing purposes. in a real world scenario, the secret key should never be exposed
  secretKey: "5ced0aeb842dcc6b071d5401987c87117229f73b1a284b7718e449f6c38fc951"
};

export const accountTwo: IAccount = {
  account: `k:5c177efe2c16d87b59dda80e4ead49abb7cfec64a99bc1535ca2070abed73d50`,
  publicKey: "5c177efe2c16d87b59dda80e4ead49abb7cfec64a99bc1535ca2070abed73d50",
  chainId: '1',
  guard: "5c177efe2c16d87b59dda80e4ead49abb7cfec64a99bc1535ca2070abed73d50s",
  // this is here only for testing purposes. in a real world scenario, the secret key should never be exposed
  secretKey: "8b454fa5aa80870dfa83dfe6a8ed739dd2291a06fdeb70380d0e1efa7c07955c"
};
