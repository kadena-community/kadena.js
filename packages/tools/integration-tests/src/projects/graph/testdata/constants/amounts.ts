export const fundAmount = 100
export const transferAmount = '20'
