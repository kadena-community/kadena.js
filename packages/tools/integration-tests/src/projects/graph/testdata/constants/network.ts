export const grapHost = 'http://localhost:4000/graphql';
export const devnetHost = 'http://localhost:8080/';
export const networkId = 'fast-development';
