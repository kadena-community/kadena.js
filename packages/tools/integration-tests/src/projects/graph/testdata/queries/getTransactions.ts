export function getTransactionsQuery(accountName: string) {
  return {
    query: `query getTransactions($accountName: String) {
      transactions(
        accountName: $accountName
      ) {
        totalCount
        edges {
          node {
            code
            data
            gas
            gasLimit
            gasPrice
            id
            senderAccount
            transfers {
              amount
              chainId
              id
              receiverAccount
              requestKey
              senderAccount
            }
            ttl
            chainId
            requestKey
            eventCount
            events {
              requestKey
              parameterText
              id
            }
            signers {
              capabilities
              id
              publicKey
              requestKey
            }
          }
        }
      }
    }`,
    variables: { accountName: accountName },
    operationName: 'getTransactions',
    extensions: {},
  };
}


export function getxChainTransactionsQuery(accountName: string) {
  return {
    query: `query getxChainTransactionsQuery($accountName: String) {
      transactions(
        accountName: $accountName
      ) {
        totalCount
        edges {
          node {
            chainId
            code
            continuation
            data
            gas
            gasLimit
            gasPrice
            id
            senderAccount
            transfers {
              amount
              chainId
              id
              receiverAccount
              requestKey
              senderAccount
            }
            step
            ttl
            requestKey
            eventCount
            events {
              chainId
              requestKey
              parameterText
              id
            }
            signers {
              capabilities
              id
              publicKey
              requestKey
            }
          }
        }
      }
    }`,
    variables: { accountName: accountName },
    operationName: 'getxChainTransactionsQuery',
    extensions: {},
  };
}
