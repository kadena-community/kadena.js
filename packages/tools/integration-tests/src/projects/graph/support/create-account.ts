import type { ICommandResult } from '@kadena/client';
import { createSignWithKeypair } from '@kadena/client';
import { transferCreate } from '@kadena/client-utils/coin';
import type { IAccount } from '../testdata/constants/accounts';
import { sender00Account } from '../testdata/constants/accounts';
import { devnetHost, networkId } from '../testdata/constants/network';

export async function createAccount(input: IAccount): Promise<ICommandResult> {
  return new Promise<ICommandResult>((resolve, reject) => {
    transferCreate(
      {
        sender: {
          account: sender00Account.account,
          publicKeys: [sender00Account.publicKey],
        },
        receiver: {
          account: input.account,
          keyset: {
            keys: [input.publicKey],
            pred: 'keys-all',
          },
        },
        amount: '100',
        chainId: input.chainId,
      },
      {
        host: devnetHost,
        defaults: {
          networkId: networkId,
        },
        sign: createSignWithKeypair([sender00Account]),
      },
    )
      .on('listen', resolve)
      .execute()
      .catch(reject);
  });
}
