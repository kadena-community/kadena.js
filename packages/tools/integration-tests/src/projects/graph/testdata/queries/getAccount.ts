import { sender00Account } from "../constants/accounts";

export const getAccountQuery = {
  query:
    'query getAccount($moduleName: String!, $accountName: String!) {\n  account(moduleName: $moduleName, accountName: $accountName) {\n    ...AllAccountFields\n    chainAccounts {\n      ...CoreChainAccountFields\n      guard {\n        keys\n        predicate\n        __typename\n      }\n      __typename\n    }\n    transactions {\n      edges {\n        node {\n          ...CoreTransactionFields\n          __typename\n        }\n        __typename\n      }\n      __typename\n    }\n    transfers {\n      edges {\n        node {\n          ...CoreTransferFields\n          crossChainTransfer {\n            ...CoreTransferFields\n            __typename\n          }\n          transaction {\n            pactId\n            __typename\n          }\n          __typename\n        }\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment AllAccountFields on ModuleAccount {\n  ...CoreAccountFields\n  id\n  totalBalance\n  __typename\n}\n\nfragment CoreAccountFields on ModuleAccount {\n  accountName\n  moduleName\n  __typename\n}\n\nfragment CoreChainAccountFields on ChainModuleAccount {\n  balance\n  chainId\n  __typename\n}\n\nfragment CoreTransactionFields on Transaction {\n  chainId\n  code\n  creationTime\n  height\n  requestKey\n  __typename\n}\n\nfragment CoreTransferFields on Transfer {\n  amount\n  chainId\n  senderAccount\n  height\n  requestKey\n  receiverAccount\n  __typename\n}',
  variables: { moduleName: 'coin', accountName: sender00Account.account },
  operationName: 'getAccount',
  extensions: {},
};
