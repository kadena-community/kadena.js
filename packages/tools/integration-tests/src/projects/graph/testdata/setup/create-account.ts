import { createSignWithKeypair } from '@kadena/client';
import { transferCreate } from '@kadena/client-utils/coin';
import { accountOne, sender00Account } from '../constants/accounts';
import { devnetHost, networkId } from '../constants/network';

export async function createAccounts(): Promise<void> {
  await transferCreate(
    {
      sender: {
        account: sender00Account.account,
        publicKeys: [sender00Account.publicKey],
      },
      receiver: {
        account: accountOne.account,
        keyset: {
          keys: [accountOne.publicKey],
          pred: 'keys-all',
        },
      },
      amount: '100',
      chainId: '0',
    },
    {
      host: devnetHost,
      defaults: {
        networkId: networkId,
      },
      sign: createSignWithKeypair([sender00Account]),
    },
  ) // signed Tx
    .on('sign', (data) => console.log(data))
    // preflight result
    .on('preflight', (data) => console.log(data))
    // submit result
    .on('submit', (data) => console.log(data))
    // listen result
    .on('listen', (data) => console.log(data))
    .execute();
}
