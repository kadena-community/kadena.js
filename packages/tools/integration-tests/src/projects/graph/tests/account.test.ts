import request from 'supertest';
import { describe, expect, test } from 'vitest';
import { accountOne, sender00Account } from '../testdata/constants/accounts';
import { grapHost } from '../testdata/constants/network';
import { getAccountQuery } from '../testdata/queries/getAccount';
import { createAccount } from '../testdata/setup/create-account';

describe('Account', () => {
  test('Query: getAccount', async () => {
    // Given a test account is created.
   await createAccount(accountOne);

    // When the getAccountQuery is executed
    const query = getAccountQuery(accountOne.account);
    const response = await request(grapHost).post('').send(query);

    //Then the statuscode should be 200 and the snapshot should match, including specific property matchers.
    expect(response.statusCode).toBe(200);
   // ToMatchSnapshot uses Jests Property Matchers, any property defined below is (literally) matched against the snashot file
   // Properties that are defined below can be a little bit more flexible to prevent having to update the snapshot on every run
    expect(response.body.data.account).toMatchSnapshot({
      chainAccounts: [
        {
          balance: expect.any(Number), //Generic Assert, to prevent updating the snapshot.
        },
      ],
      totalBalance: expect.any(Number), //Generic Assert, to prevent updating the snapshot.
      transfers: {
        edges: expect.arrayContaining([
          expect.objectContaining({
            __typename: 'ModuleAccountTransfersConnectionEdge',
            node: {
              __typename: 'Transfer',
              amount: 100,
              chainId: 0,
              crossChainTransfer: null,
              height: expect.any(Number), //Generic Assert, to prevent updating the snapshot.
              receiverAccount: accountOne.account,
              requestKey: expect.any(String), //Generic Assert, to prevent updating the snapshot.
              senderAccount: sender00Account.account,
              transaction: {
                __typename: 'Transaction',
                pactId: null,
              },
            },
          }),
        ]),
      },
    });
  });
});
