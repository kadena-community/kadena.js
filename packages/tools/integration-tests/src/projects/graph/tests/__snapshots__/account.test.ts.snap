// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Account > Query: getAccount 1`] = `
{
  "__typename": "ModuleAccount",
  "accountName": "k:bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8",
  "chainAccounts": [
    {
      "__typename": "ChainModuleAccount",
      "balance": Any<Number>,
      "chainId": "0",
      "guard": {
        "__typename": "Guard",
        "keys": [
          "bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8",
        ],
        "predicate": "keys-all",
      },
    },
  ],
  "id": "TW9kdWxlQWNjb3VudDpNb2R1bGVBY2NvdW50L2NvaW4vazpiYzdiYzZlMTgwNTUxZTQyZjgzMWJlZDA4Y2E4MmMxMzdlYWQ3YThhMmIxNzQ3MjRjYTkxN2E3M2U2MmU0ZWI4",
  "moduleName": "coin",
  "totalBalance": Any<Number>,
  "transactions": {
    "__typename": "ModuleAccountTransactionsConnection",
    "edges": [],
  },
  "transfers": {
    "__typename": "ModuleAccountTransfersConnection",
    "edges": ArrayContaining [
      ObjectContaining {
        "__typename": "ModuleAccountTransfersConnectionEdge",
        "node": {
          "__typename": "Transfer",
          "amount": 100,
          "chainId": 0,
          "crossChainTransfer": null,
          "height": Any<Number>,
          "receiverAccount": "k:bc7bc6e180551e42f831bed08ca82c137ead7a8a2b174724ca917a73e62e4eb8",
          "requestKey": Any<String>,
          "senderAccount": "sender00",
          "transaction": {
            "__typename": "Transaction",
            "pactId": null,
          },
        },
      },
    ],
  },
}
`;
